### PR TITLE
feat(replica/resize): api for resizing a replica

### PR DESCRIPTION
**Pull request by dsharma-dc**

### CODE REVIEW

**Review by niladrih — APPROVED** [review]
LGTM, but still needs a second review IMO. Needs a rebase.

**Review by abhilashshetty04 — APPROVED** [review]

**Review by Abhinandan-Purkait — APPROVED** [review]

File: apis/io-engine/protobuf/v1/replica.proto
Comment (Abhinandan-Purkait) [inline apis/io-engine/protobuf/v1/replica.proto:82]: nit: shrink might not be a use case here. So maybe remove it just to be clear with the expectations?

Comment (dsharma-dc) [reply apis/io-engine/protobuf/v1/replica.proto:82]: Since this is the interface between control and data plane, we might as well leverage shrink for something internally.

**Comment by dsharma-dc** [comment]
@tiagolobocastro @Abhinandan-Purkait Apparently I'm not authorised to merge in this repo. Could you merge this please?
I'll sort out my permissions.

**Comment by Abhinandan-Purkait** [comment]
> @tiagolobocastro @Abhinandan-Purkait Apparently I'm not authorised to merge in this repo. Could you merge this please? I'll sort out my permissions.

Ohh seems like I too don't have any permissions on this repo.

**Comment by tiagolobocastro** [comment]
We're branching off develop soon (for release), so I'd say hold this up until then?

**Comment by dsharma-dc** [comment]
> We're branching off develop soon (for release), so I'd say hold this up until then?

@tiagolobocastro Sure, we can hold if that's what we've been following. Otherwise it's a harmless new protobuf so nothing breaking, and we can branch off the commits prior to this.

File: apis/io-engine/protobuf/v1/replica.proto
Comment (tiagolobocastro) [inline apis/io-engine/protobuf/v1/replica.proto]: Could we return Replica directly?
See Share/Unshare

Comment (dsharma-dc) [reply apis/io-engine/protobuf/v1/replica.proto]: > Could we return Replica directly? See Share/Unshare

I had two minds about doing that. I find this wrapped response more intuitive. In fact almost all of the response messages in Nexus protobuf follow this way.

Comment (tiagolobocastro) [reply apis/io-engine/protobuf/v1/replica.proto]: Yeah I know about the nexus that's I suggest, because we have to unwrap nexus on the client side when handling response.
Also every rpc other than list returns Replica so it'd be good to stay consistent as well?

Comment (dsharma-dc) [reply apis/io-engine/protobuf/v1/replica.proto]: Alright, updated to return `Replica` directly. PTAL

**Review by tiagolobocastro — APPROVED** [review]
Please fix linter, and remember not to merge until dataplane PR is approved.

**Comment by dsharma-dc** [comment]
> Please fix linter, and remember not to merge until dataplane PR is approved.

linter fixed, PTAL at https://github.com/openebs/mayastor/pull/1550

**Comment by dsharma-dc** [comment]
@tiagolobocastro Could you please merge this if review on mayastor/1550 is complete.

**Pull request merged**
